Sources/ScipioKit/SwiftPM/PinState.swift

```diff
@@ -0,0 +1,85 @@
+// ===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2014-2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// ===----------------------------------------------------------------------===//
+//
+// NOTE: This file has been modified from the original version.
+//
+// ===----------------------------------------------------------------------===//
+
+public enum SPMPinState: Equatable, Sendable, CustomStringConvertible {
```

Comment:
I feel it's enough just to be `PinState`

Reply:
> To keep cache compatibility, I didn’t use `Pin.State` in `Sources/Resolver/ResolvedModels.swift`. Instead, I used `PinsStore.PinState` from SwiftPM and redefined it as `SPMPinState` in `Sources/SwiftPM/PinState.swift`.

If Scipio version is changed (updated) cache keys are invalidated anyway so we don't need to such compatibility I think 👀

Reply:
Yes, as ikesyo-san said, we needn't consider backward compatibility. All caches will expire